diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jccoefct.c dcmtk-3.6.7/dcmjpeg/libijg12/jccoefct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jccoefct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jccoefct.c	2022-08-16 12:21:34.000000000 +0200
@@ -343,7 +343,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jcdiffct.c dcmtk-3.6.7/dcmjpeg/libijg12/jcdiffct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jcdiffct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jcdiffct.c	2022-08-16 12:21:20.000000000 +0200
@@ -302,7 +302,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossless_c_ptr losslsc = (j_lossless_c_ptr) cinfo->codec;
   c_diff_ptr diff = (c_diff_ptr) losslsc->diff_private;
   /* JDIMENSION MCU_col_num; */ /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jcpred.c dcmtk-3.6.7/dcmjpeg/libijg12/jcpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jcpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jcpred.c	2022-08-16 12:21:04.000000000 +0200
@@ -213,7 +213,7 @@
          const JSAMPROW input_buf, JSAMPROW prev_row,
          JDIFFROW diff_buf, JDIMENSION width)
 {
-  (void)prev_row;
+  //(void)prev_row;
   DIFFERENCE_1D(INITIAL_PREDICTORx);
 
   /*
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jctrans.c dcmtk-3.6.7/dcmjpeg/libijg12/jctrans.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jctrans.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jctrans.c	2022-08-16 12:20:36.000000000 +0200
@@ -267,7 +267,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdmerge.c dcmtk-3.6.7/dcmjpeg/libijg12/jdmerge.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdmerge.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jdmerge.c	2022-08-16 12:20:14.000000000 +0200
@@ -148,7 +148,7 @@
             JDIMENSION out_rows_avail)
 /* 2:1 vertical sampling case: may need a spare row. */
 {
-  (void) in_row_groups_avail;
+  //(void) in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   JSAMPROW work_ptrs[2];
   JDIMENSION num_rows;      /* number of rows returned to caller */
@@ -198,8 +198,8 @@
             JDIMENSION out_rows_avail)
 /* 1:1 vertical sampling case: much easier, never need a spare row. */
 {
-  (void) in_row_groups_avail;
-  (void) out_rows_avail;
+  //(void) in_row_groups_avail;
+  //(void) out_rows_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
 
   /* Just do the upsampling. */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdpostct.c dcmtk-3.6.7/dcmjpeg/libijg12/jdpostct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdpostct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jdpostct.c	2022-08-16 12:19:54.000000000 +0200
@@ -161,8 +161,8 @@
               JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
               JDIMENSION out_rows_avail)
 {
-  (void) output_buf;
-  (void) out_rows_avail;
+  //(void) output_buf;
+  //(void) out_rows_avail;
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION old_next_row, num_rows;
 
@@ -207,9 +207,9 @@
             JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
             JDIMENSION out_rows_avail)
 {
-  (void) input_buf;
-  (void) in_row_group_ctr;
-  (void) in_row_groups_avail;
+  //(void) input_buf;
+  //(void) in_row_group_ctr;
+  //(void) in_row_groups_avail;
 
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION num_rows, max_rows;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdpred.c dcmtk-3.6.7/dcmjpeg/libijg12/jdpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jdpred.c	2022-08-16 12:19:22.000000000 +0200
@@ -101,8 +101,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_1D(INITIAL_PREDICTOR2);
 }
 
@@ -111,8 +111,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR2);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -123,8 +123,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR3);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -135,8 +135,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR4);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -147,8 +147,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR5);
   JPEG_UNUSED(Rc);
@@ -160,8 +160,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR6);
   JPEG_UNUSED(Rc);
@@ -173,8 +173,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR7);
   JPEG_UNUSED(Rc);
@@ -195,7 +195,7 @@
                 JDIFFROW undiff_buf, JDIMENSION width)
 {
 
-  (void)prev_row;
+  //(void)prev_row;
   j_lossless_d_ptr losslsd = (j_lossless_d_ptr) cinfo->codec;
 
   UNDIFFERENCE_1D(INITIAL_PREDICTORx);
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdsample.c dcmtk-3.6.7/dcmjpeg/libijg12/jdsample.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdsample.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jdsample.c	2022-08-16 12:18:32.000000000 +0200
@@ -92,7 +92,7 @@
           JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
           JDIMENSION out_rows_avail)
 {
-  (void)in_row_groups_avail;
+  //(void)in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   int ci;
   jpeg_component_info * compptr;
@@ -239,7 +239,7 @@
 h2v1_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
@@ -268,7 +268,7 @@
 h2v2_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdscale.c dcmtk-3.6.7/dcmjpeg/libijg12/jdscale.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jdscale.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jdscale.c	2022-08-16 12:18:02.000000000 +0200
@@ -67,7 +67,7 @@
 	const JDIFFROW diff_buf, JSAMPROW output_buf,
 	JDIMENSION width)
 {
-  (void)cinfo;
+  //(void)cinfo;
   unsigned int xindex;
 
   for (xindex = 0; xindex < width; xindex++)
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jquant1.c dcmtk-3.6.7/dcmjpeg/libijg12/jquant1.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jquant1.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jquant1.c	2022-08-16 12:17:44.000000000 +0200
@@ -744,7 +744,7 @@
 METHODDEF(void)
 start_pass_1_quant (j_decompress_ptr cinfo, boolean is_pre_scan)
 {
-  (void) is_pre_scan;
+  //(void) is_pre_scan;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   size_t arraysize;
   int i;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg12/jquant2.c dcmtk-3.6.7/dcmjpeg/libijg12/jquant2.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg12/jquant2.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg12/jquant2.c	2022-08-16 12:17:30.000000000 +0200
@@ -224,7 +224,7 @@
 prescan_quantize (j_decompress_ptr cinfo, JSAMPARRAY input_buf,
           JSAMPARRAY output_buf, int num_rows)
 {
-  (void) output_buf;
+  //(void) output_buf;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   register JSAMPROW ptr;
   register histptr histp;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jccoefct.c dcmtk-3.6.7/dcmjpeg/libijg16/jccoefct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jccoefct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jccoefct.c	2022-08-16 12:17:02.000000000 +0200
@@ -343,7 +343,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jcdiffct.c dcmtk-3.6.7/dcmjpeg/libijg16/jcdiffct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jcdiffct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jcdiffct.c	2022-08-16 12:14:16.000000000 +0200
@@ -302,7 +302,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossless_c_ptr losslsc = (j_lossless_c_ptr) cinfo->codec;
   c_diff_ptr diff = (c_diff_ptr) losslsc->diff_private;
   /* JDIMENSION MCU_col_num; */ /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jcpred.c dcmtk-3.6.7/dcmjpeg/libijg16/jcpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jcpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jcpred.c	2022-08-16 12:14:00.000000000 +0200
@@ -213,7 +213,7 @@
          const JSAMPROW input_buf, JSAMPROW prev_row,
          JDIFFROW diff_buf, JDIMENSION width)
 {
-  (void)prev_row;
+  //(void)prev_row;
   DIFFERENCE_1D(INITIAL_PREDICTORx);
 
   /*
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jctrans.c dcmtk-3.6.7/dcmjpeg/libijg16/jctrans.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jctrans.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jctrans.c	2022-08-16 12:13:42.000000000 +0200
@@ -267,7 +267,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdmerge.c dcmtk-3.6.7/dcmjpeg/libijg16/jdmerge.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdmerge.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jdmerge.c	2022-08-16 12:13:18.000000000 +0200
@@ -171,7 +171,7 @@
             JDIMENSION out_rows_avail)
 /* 2:1 vertical sampling case: may need a spare row. */
 {
-  (void) in_row_groups_avail;
+  //(void) in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   JSAMPROW work_ptrs[2];
   JDIMENSION num_rows;      /* number of rows returned to caller */
@@ -221,8 +221,8 @@
             JDIMENSION out_rows_avail)
 /* 1:1 vertical sampling case: much easier, never need a spare row. */
 {
-  (void) in_row_groups_avail;
-  (void) out_rows_avail;
+  //(void) in_row_groups_avail;
+  //(void) out_rows_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
 
   /* Just do the upsampling. */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdpostct.c dcmtk-3.6.7/dcmjpeg/libijg16/jdpostct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdpostct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jdpostct.c	2022-08-16 12:12:54.000000000 +0200
@@ -161,8 +161,8 @@
               JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
               JDIMENSION out_rows_avail)
 {
-  (void) output_buf;
-  (void) out_rows_avail;
+  //(void) output_buf;
+  //(void) out_rows_avail;
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION old_next_row, num_rows;
 
@@ -207,9 +207,9 @@
             JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
             JDIMENSION out_rows_avail)
 {
-  (void) input_buf;
-  (void) in_row_group_ctr;
-  (void) in_row_groups_avail;
+  //(void) input_buf;
+  //(void) in_row_group_ctr;
+  //(void) in_row_groups_avail;
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION num_rows, max_rows;
 
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdpred.c dcmtk-3.6.7/dcmjpeg/libijg16/jdpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jdpred.c	2022-08-16 12:12:26.000000000 +0200
@@ -101,8 +101,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_1D(INITIAL_PREDICTOR2);
 }
 
@@ -111,8 +111,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-   (void)cinfo;
-  (void)comp_index;
+   //(void)cinfo;
+  //(void)comp_index;
  UNDIFFERENCE_2D(PREDICTOR2);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -123,8 +123,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR3);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -135,8 +135,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR4A);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -147,8 +147,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR4);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -159,8 +159,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR5);
   JPEG_UNUSED(Rc);
@@ -172,8 +172,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR5A);
   JPEG_UNUSED(Rc);
@@ -185,8 +185,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR6);
   JPEG_UNUSED(Rc);
@@ -198,8 +198,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR6A);
   JPEG_UNUSED(Rc);
@@ -211,8 +211,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR7);
   JPEG_UNUSED(Rc);
@@ -224,8 +224,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR7A);
   JPEG_UNUSED(Rc);
@@ -245,7 +245,7 @@
                 const JDIFFROW diff_buf, JDIFFROW prev_row,
                 JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)prev_row;
+  //(void)prev_row;
   j_lossless_d_ptr losslsd = (j_lossless_d_ptr) cinfo->codec;
 
   UNDIFFERENCE_1D(INITIAL_PREDICTORx);
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdsample.c dcmtk-3.6.7/dcmjpeg/libijg16/jdsample.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdsample.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jdsample.c	2022-08-16 12:10:32.000000000 +0200
@@ -92,7 +92,7 @@
           JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
           JDIMENSION out_rows_avail)
 {
-  (void)in_row_groups_avail;
+  //(void)in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   int ci;
   jpeg_component_info * compptr;
@@ -239,7 +239,7 @@
 h2v1_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
@@ -268,7 +268,7 @@
 h2v2_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdscale.c dcmtk-3.6.7/dcmjpeg/libijg16/jdscale.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jdscale.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jdscale.c	2022-08-16 12:04:18.000000000 +0200
@@ -67,8 +67,8 @@
 	const JDIFFROW diff_buf, JSAMPROW output_buf,
 	JDIMENSION width)
 {
-  (void)cinfo;
   unsigned int xindex;
+  (void)cinfo;
 
   for (xindex = 0; xindex < width; xindex++)
     output_buf[xindex] = (JSAMPLE) diff_buf[xindex];
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jquant1.c dcmtk-3.6.7/dcmjpeg/libijg16/jquant1.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jquant1.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jquant1.c	2022-08-16 12:03:56.000000000 +0200
@@ -744,10 +744,10 @@
 METHODDEF(void)
 start_pass_1_quant (j_decompress_ptr cinfo, boolean is_pre_scan)
 {
-  (void) is_pre_scan;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   size_t arraysize;
   int i;
+  (void) is_pre_scan;
 
   /* Install my colormap. */
   cinfo->colormap = cquantize->sv_colormap;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg16/jquant2.c dcmtk-3.6.7/dcmjpeg/libijg16/jquant2.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg16/jquant2.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg16/jquant2.c	2022-08-16 12:03:14.000000000 +0200
@@ -224,7 +224,6 @@
 prescan_quantize (j_decompress_ptr cinfo, JSAMPARRAY input_buf,
           JSAMPARRAY output_buf, int num_rows)
 {
-  (void) output_buf;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   register JSAMPROW ptr;
   register histptr histp;
@@ -232,6 +231,7 @@
   int row;
   JDIMENSION col;
   JDIMENSION width = cinfo->output_width;
+  (void) output_buf;
 
   for (row = 0; row < num_rows; row++) {
     ptr = input_buf[row];
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jccoefct.c dcmtk-3.6.7/dcmjpeg/libijg8/jccoefct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jccoefct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jccoefct.c	2022-08-16 12:27:04.000000000 +0200
@@ -343,7 +343,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jcdiffct.c dcmtk-3.6.7/dcmjpeg/libijg8/jcdiffct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jcdiffct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jcdiffct.c	2022-08-16 12:26:48.000000000 +0200
@@ -302,7 +302,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossless_c_ptr losslsc = (j_lossless_c_ptr) cinfo->codec;
   c_diff_ptr diff = (c_diff_ptr) losslsc->diff_private;
   /* JDIMENSION MCU_col_num; */ /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jcpred.c dcmtk-3.6.7/dcmjpeg/libijg8/jcpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jcpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jcpred.c	2022-08-16 12:26:32.000000000 +0200
@@ -213,7 +213,7 @@
          const JSAMPROW input_buf, JSAMPROW prev_row,
          JDIFFROW diff_buf, JDIMENSION width)
 {
-  (void)prev_row;
+  //(void)prev_row;
   DIFFERENCE_1D(INITIAL_PREDICTORx);
 
   /*
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jctrans.c dcmtk-3.6.7/dcmjpeg/libijg8/jctrans.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jctrans.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jctrans.c	2022-08-16 12:25:56.000000000 +0200
@@ -267,7 +267,7 @@
 METHODDEF(boolean)
 compress_output (j_compress_ptr cinfo, JSAMPIMAGE input_buf)
 {
-  (void)input_buf;
+  //(void)input_buf;
   j_lossy_c_ptr lossyc = (j_lossy_c_ptr) cinfo->codec;
   c_coef_ptr coef = (c_coef_ptr) lossyc->coef_private;
   JDIMENSION MCU_col_num;   /* index of current MCU within row */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdmerge.c dcmtk-3.6.7/dcmjpeg/libijg8/jdmerge.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdmerge.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jdmerge.c	2022-08-16 12:25:36.000000000 +0200
@@ -148,7 +148,7 @@
             JDIMENSION out_rows_avail)
 /* 2:1 vertical sampling case: may need a spare row. */
 {
-  (void) in_row_groups_avail;
+  //(void) in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   JSAMPROW work_ptrs[2];
   JDIMENSION num_rows;      /* number of rows returned to caller */
@@ -198,8 +198,8 @@
             JDIMENSION out_rows_avail)
 /* 1:1 vertical sampling case: much easier, never need a spare row. */
 {
-  (void) in_row_groups_avail;
-  (void) out_rows_avail;
+  //(void) in_row_groups_avail;
+  //(void) out_rows_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
 
   /* Just do the upsampling. */
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdpostct.c dcmtk-3.6.7/dcmjpeg/libijg8/jdpostct.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdpostct.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jdpostct.c	2022-08-16 12:25:12.000000000 +0200
@@ -161,8 +161,8 @@
               JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
               JDIMENSION out_rows_avail)
 {
-  (void) output_buf;
-  (void) out_rows_avail;
+  //(void) output_buf;
+  //(void) out_rows_avail;
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION old_next_row, num_rows;
 
@@ -207,9 +207,9 @@
             JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
             JDIMENSION out_rows_avail)
 {
-  (void) input_buf;
-  (void) in_row_group_ctr;
-  (void) in_row_groups_avail;
+  //(void) input_buf;
+  //(void) in_row_group_ctr;
+  //(void) in_row_groups_avail;
   my_post_ptr post = (my_post_ptr) cinfo->post;
   JDIMENSION num_rows, max_rows;
 
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdpred.c dcmtk-3.6.7/dcmjpeg/libijg8/jdpred.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdpred.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jdpred.c	2022-08-16 12:23:34.000000000 +0200
@@ -101,8 +101,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_1D(INITIAL_PREDICTOR2);
 }
 
@@ -111,8 +111,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR2);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -123,8 +123,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR3);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -135,8 +135,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   UNDIFFERENCE_2D(PREDICTOR4);
   JPEG_UNUSED(Rc);
   JPEG_UNUSED(Rb);
@@ -147,8 +147,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR5);
   JPEG_UNUSED(Rc);
@@ -160,8 +160,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR6);
   JPEG_UNUSED(Rc);
@@ -173,8 +173,8 @@
            const JDIFFROW diff_buf, const JDIFFROW prev_row,
            JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)cinfo;
-  (void)comp_index;
+  //(void)cinfo;
+  //(void)comp_index;
   SHIFT_TEMPS
   UNDIFFERENCE_2D(PREDICTOR7);
   JPEG_UNUSED(Rc);
@@ -194,7 +194,7 @@
                 const JDIFFROW diff_buf, JDIFFROW prev_row,
                 JDIFFROW undiff_buf, JDIMENSION width)
 {
-  (void)prev_row;
+  //(void)prev_row;
   j_lossless_d_ptr losslsd = (j_lossless_d_ptr) cinfo->codec;
 
   UNDIFFERENCE_1D(INITIAL_PREDICTORx);
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdsample.c dcmtk-3.6.7/dcmjpeg/libijg8/jdsample.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdsample.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jdsample.c	2022-08-16 12:22:36.000000000 +0200
@@ -92,7 +92,7 @@
           JSAMPARRAY output_buf, JDIMENSION *out_row_ctr,
           JDIMENSION out_rows_avail)
 {
-  (void)in_row_groups_avail;
+  //(void)in_row_groups_avail;
   my_upsample_ptr upsample = (my_upsample_ptr) cinfo->upsample;
   int ci;
   jpeg_component_info * compptr;
@@ -239,7 +239,7 @@
 h2v1_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
@@ -268,7 +268,7 @@
 h2v2_upsample (j_decompress_ptr cinfo, jpeg_component_info * compptr,
            JSAMPARRAY input_data, JSAMPARRAY * output_data_ptr)
 {
-  (void)compptr;
+  //(void)compptr;
   JSAMPARRAY output_data = *output_data_ptr;
   register JSAMPROW inptr, outptr;
   register JSAMPLE invalue;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdscale.c dcmtk-3.6.7/dcmjpeg/libijg8/jdscale.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jdscale.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jdscale.c	2022-08-16 12:22:06.000000000 +0200
@@ -67,7 +67,7 @@
 	const JDIFFROW diff_buf, JSAMPROW output_buf,
 	JDIMENSION width)
 {
-  (void)cinfo;
+  //(void)cinfo;
   unsigned int xindex;
 
   for (xindex = 0; xindex < width; xindex++)
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jquant1.c dcmtk-3.6.7/dcmjpeg/libijg8/jquant1.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jquant1.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jquant1.c	2022-08-16 12:21:50.000000000 +0200
@@ -744,7 +744,7 @@
 METHODDEF(void)
 start_pass_1_quant (j_decompress_ptr cinfo, boolean is_pre_scan)
 {
-  (void) is_pre_scan;
+  //(void) is_pre_scan;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   size_t arraysize;
   int i;
diff -urEb dcmtk-3.6.7.orig/dcmjpeg/libijg8/jquant2.c dcmtk-3.6.7/dcmjpeg/libijg8/jquant2.c
--- dcmtk-3.6.7.orig/dcmjpeg/libijg8/jquant2.c	2022-04-28 15:47:25.000000000 +0200
+++ dcmtk-3.6.7/dcmjpeg/libijg8/jquant2.c	2022-08-16 12:03:36.000000000 +0200
@@ -224,7 +224,6 @@
 prescan_quantize (j_decompress_ptr cinfo, JSAMPARRAY input_buf,
           JSAMPARRAY output_buf, int num_rows)
 {
-  (void) output_buf;
   my_cquantize_ptr cquantize = (my_cquantize_ptr) cinfo->cquantize;
   register JSAMPROW ptr;
   register histptr histp;
@@ -232,6 +231,7 @@
   int row;
   JDIMENSION col;
   JDIMENSION width = cinfo->output_width;
+  (void) output_buf;
 
   for (row = 0; row < num_rows; row++) {
     ptr = input_buf[row];
