diff -urEb civetweb-1.11.orig/src/civetweb.c civetweb-1.11/src/civetweb.c
--- civetweb-1.11.orig/src/civetweb.c	2019-01-16 10:52:21.313348500 +0100
+++ civetweb-1.11/src/civetweb.c	2019-01-16 11:59:58.995605655 +0100
@@ -20,6 +20,14 @@
  * THE SOFTWARE.
  */
 
+
+#if defined(__LSB_VERSION__)
+// These definitions are mandatory for Linux Standard Base
+#  define NO_CACHING
+#  define NO_ALTERNATIVE_QUEUE
+#  define NO_THREAD_NAME
+#endif
+  
 #if defined(__GNUC__) || defined(__MINGW32__)
 #define GCC_VERSION                                                            \
 	(__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
@@ -536,6 +544,10 @@
 #if !defined(EWOULDBLOCK)
 #define EWOULDBLOCK WSAEWOULDBLOCK
 #endif /* !EWOULDBLOCK */
+#if !defined(ECONNRESET)
+// Macro ECONNRESET is not always defined (e.g. on Visual Studio 2008)
+#  define ECONNRESET WSAECONNRESET
+#endif /* !ECONNRESET */
 #define _POSIX_
 #define INT64_FMT "I64d"
 #define UINT64_FMT "I64u"
