diff -urEb dcmtk-3.6.1_20170228.orig/dcmnet/libsrc/dul.cc dcmtk-3.6.1_20170228/dcmnet/libsrc/dul.cc
--- dcmtk-3.6.1_20170228.orig/dcmnet/libsrc/dul.cc	2017-07-05 12:57:20.707214499 +0200
+++ dcmtk-3.6.1_20170228/dcmnet/libsrc/dul.cc	2017-07-05 12:58:03.563590489 +0200
@@ -1789,7 +1789,7 @@
         return makeDcmnetCondition(DULC_TCPINITERROR, OF_error, msg.c_str());
     }
 #endif
-    setTCPBufferLength(sock);
+    //setTCPBufferLength(sock);
 
     /*
      * Disable the so-called Nagle algorithm (if requested).
diff -urEb dcmtk-3.6.1_20170228.orig/dcmnet/libsrc/dulfsm.cc dcmtk-3.6.1_20170228/dcmnet/libsrc/dulfsm.cc
--- dcmtk-3.6.1_20170228.orig/dcmnet/libsrc/dulfsm.cc	2017-07-05 12:57:20.707214499 +0200
+++ dcmtk-3.6.1_20170228/dcmnet/libsrc/dulfsm.cc	2017-07-05 12:58:17.995717258 +0200
@@ -2419,7 +2419,7 @@
           return makeDcmnetCondition(DULC_TCPINITERROR, OF_error, msg.c_str());
         }
 #endif
-        setTCPBufferLength(s);
+        //setTCPBufferLength(s);
 
         /*
          * Disable the so-called Nagle algorithm (if requested).
