diff -urEb dcmtk-3.6.2.orig/dcmnet/libsrc/dul.cc dcmtk-3.6.2/dcmnet/libsrc/dul.cc
--- dcmtk-3.6.2.orig/dcmnet/libsrc/dul.cc	2017-07-17 17:54:10.406140454 +0200
+++ dcmtk-3.6.2/dcmnet/libsrc/dul.cc	2017-07-17 17:54:48.681341404 +0200
@@ -1862,7 +1862,7 @@
         msg += OFStandard::getLastNetworkErrorCode().message();
         return makeDcmnetCondition(DULC_TCPINITERROR, OF_error, msg.c_str());
     }
-    setTCPBufferLength(sock);
+    //setTCPBufferLength(sock);
 
     /*
      * Disable the so-called Nagle algorithm (if requested).
diff -urEb dcmtk-3.6.2.orig/dcmnet/libsrc/dulfsm.cc dcmtk-3.6.2/dcmnet/libsrc/dulfsm.cc
--- dcmtk-3.6.2.orig/dcmnet/libsrc/dulfsm.cc	2017-07-17 17:54:10.406140454 +0200
+++ dcmtk-3.6.2/dcmnet/libsrc/dulfsm.cc	2017-07-17 17:55:04.122634664 +0200
@@ -2447,7 +2447,7 @@
           msg += OFStandard::getLastNetworkErrorCode().message();
           return makeDcmnetCondition(DULC_TCPINITERROR, OF_error, msg.c_str());
         }
-        setTCPBufferLength(s);
+        //setTCPBufferLength(s);
 
         /*
          * Disable the so-called Nagle algorithm (if requested).
