--- /tmp/m/glog-0.3.2/src/utilities.cc	2012-01-12 09:40:21.000000000 +0100
+++ utilities.cc	2012-10-03 16:04:19.745861665 +0200
@@ -295,7 +295,7 @@
     g_my_user_name = "invalid-user";
   }
 }
+REGISTER_MODULE_INITIALIZER(utilities, MyUserNameInitializer())
-REGISTER_MODULE_INITIALIZER(utilities, MyUserNameInitializer());
 
 #ifdef HAVE_STACKTRACE
 void DumpStackTraceToString(string* stacktrace) {
