diff -urEb libp11-0.4.0.orig/src/atfork.c libp11-0.4.0/src/atfork.c
--- libp11-0.4.0.orig/src/atfork.c	2020-03-05 20:48:55.447852662 +0100
+++ libp11-0.4.0/src/atfork.c	2020-03-05 20:49:05.983770656 +0100
@@ -25,7 +25,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
-#include <atfork.h>
+#include "atfork.h"
 
 #ifdef __sun
 # pragma fini(lib_deinit)
diff -urEb libp11-0.4.0.orig/src/engine.h libp11-0.4.0/src/engine.h
--- libp11-0.4.0.orig/src/engine.h	2020-03-05 20:48:55.447852662 +0100
+++ libp11-0.4.0/src/engine.h	2020-03-05 20:49:05.983770656 +0100
@@ -29,7 +29,7 @@
 #define _ENGINE_PKCS11_H
 
 #ifndef _WIN32
-#include "config.h"
+//#include "config.h"
 #endif
 
 #include "libp11.h"
diff -urEb libp11-0.4.0.orig/src/libp11-int.h libp11-0.4.0/src/libp11-int.h
--- libp11-0.4.0.orig/src/libp11-int.h	2020-03-05 20:48:55.447852662 +0100
+++ libp11-0.4.0/src/libp11-int.h	2020-03-05 20:49:05.983770656 +0100
@@ -20,7 +20,7 @@
 #define _LIBP11_INT_H
 
 #ifndef _WIN32
-#include "config.h"
+//#include "config.h"
 #endif
 
 #include "libp11.h"
diff -urEb libp11-0.4.0.orig/src/p11_key.c libp11-0.4.0/src/p11_key.c
--- libp11-0.4.0.orig/src/p11_key.c	2020-03-05 20:48:55.447852662 +0100
+++ libp11-0.4.0/src/p11_key.c	2020-03-05 20:49:24.959625180 +0100
@@ -21,6 +21,10 @@
 #include <string.h>
 #include <openssl/bn.h>
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L // OpenSSL 1.0.2
+#  include <crypto/rsa/rsa_locl.h>
+#endif
+
 #ifdef _WIN32
 #define strncasecmp strnicmp
 #endif
diff -urEb libp11-0.4.0.orig/src/p11_rsa.c libp11-0.4.0/src/p11_rsa.c
--- libp11-0.4.0.orig/src/p11_rsa.c	2020-03-05 20:48:55.447852662 +0100
+++ libp11-0.4.0/src/p11_rsa.c	2020-03-05 20:49:20.095662204 +0100
@@ -27,6 +27,10 @@
 #include <openssl/evp.h>
 #include <openssl/rsa.h>
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L // OpenSSL 1.0.2
+#  include <crypto/rsa/rsa_locl.h>
+#endif
+
 static int rsa_ex_index = 0;
 
 #if OPENSSL_VERSION_NUMBER < 0x10100003L
